pkg/operator/revisionpoddeployer/oauthrevisionpoddeployer.go

```diff
@@ -0,0 +1,31 @@
+package revisionpoddeployer
+
+import (
+	"github.com/openshift/library-go/pkg/operator/encryption/statemachine"
+)
+
+type oauthRevisionPodDeployer struct {
+	statemachine.Deployer
+	isOAuthEncryptionConfigManagedByThisOperatorFunc func() bool
+}
+
+var _ DeployerExtended = &oauthRevisionPodDeployer{}
+
+// NewOAuthRevisionLabelPodDeployerAdapter returns a deployer used by the encryption controllers.
+// Whether this deployer is on/off is determined by isOAuthEncryptionConfigManagedByThisOperatorFunc
+// TODO:
+//  remove this deployer in 4.7
+func NewOAuthRevisionLabelPodDeployerAdapter(
```

Comment:
MaybeDisabledDeployer might be a better name? This is nothing special about oauth.

Reply:
DisabledByPredicateDeployer even. And the interface would be MaybeDisabledDeployer.